### PR TITLE
Add decorated function to pytest runnables

### LINKED ISSUE #22324: Allow decorated_function in pytest-based test discovery and runnables for Python
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

In a pytest-based Python project, the `@pytest.mark` decorator is often used to mark or parametrize tests. Those tests should also be discoverable and runnable with `task ` commands.

### Zed Version and System Specs

```
Zed: v0.167.1 (Zed Preview)
OS: macOS 14.5.0
Memory: 32 GiB
Architecture: aarch64
```

### If applicable, add screenshots or screencasts of the incorrect state / behavior

CleanShot 2024-12-21 at 10 02 38@2x

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

```

```
</details>

### PR DESCRIPTION
**Pull request by ThomAub**
Fixes: #22324

Closes #ISSUE

Release Notes:

- Fixed pytest decoracted function discovery

### CODE REVIEW

**Comment by ThomAub** [comment]
CleanShot 2024-12-21 at 10 03 00@2x

**Comment by osiewicz** [comment]
Thanks!

File: crates/languages/src/python/runnables.scm [hunk 1]
```diff
@@ -41,6 +41,20 @@
     (#set! tag python-pytest-method)
 )
 
+; decorated pytest functions
+(
+    (module
+        (decorated_definition
+            (decorator)+ @decorator
+            definition: (function_definition
+                name: (identifier) @run @_pytest_decorated_method_name
```
Comment (osiewicz) [inline crates/languages/src/python/runnables.scm:50]: I've just noticed (after the merge :facepalm:) that you've used a different capture name here; I'm pretty sure it won't spawn any tasks? (Even though it'll show up in a gutter)
I'd suggest just using: `@_pytest_method_name`

File: crates/languages/src/python/runnables.scm [hunk 2]
```diff
@@ -41,6 +41,20 @@
     (#set! tag python-pytest-method)
 )
 
+; decorated pytest functions
+(
+    (module
+        (decorated_definition
+            (decorator)+ @decorator
```
Comment (osiewicz) [inline crates/languages/src/python/runnables.scm:48]: This should prolly be `@_decorator`

**Pull request merged**